generator/index.ts

```diff
@@ -1,308 +1,3 @@
-          if (values.items.items["$ref"]) {
-            return `${getRefString(values.items.items["$ref"])}[][]`;
-          }
-          return "[]";
-        }
-        return `${values.items.type}[]`;
-      }
-      if (values.type === "integer") {
-        return "number";
-      }
-      return values.type;
-    }
-    if (values["$ref"]) {
-      return getRefString(values["$ref"]);
-    }
-    return "any"; // If we don't know the type then type it as any.
-  }
-
-  function isRequired(requiredArgs: string[], propertyName: string): boolean {
-    if (!requiredArgs.length) {
-      // If requiredArgs doesn't exist then it's likely that this is a
-      // response interface.
-      return true;
-    }
-    return requiredArgs.includes(propertyName);
-  }
-
-  return Object.entries(properties).map((property) => {
-    return {
-      name: property[0],
-      type: extractType(property[1]),
-      required: isRequired(required, property[0]),
-    };
-  });
-}
-
-function generateFile(facadeTemplateData: FacadeTemplate): void {
-  const output: string = generateFacadeTemplate(facadeTemplateData);
-  const filename = `${facadeTemplateData.name}V${facadeTemplateData.version}`;
-  const facadeFoldername = facadeFolderName(facadeTemplateData.name);
-
-  const outputFolder = `api/facades/${facadeFoldername}/`;
-  mkdirSync(outputFolder, { recursive: true });
-  const newFacadeFilePath = join(outputFolder, `${filename}.ts`);
-  // Avoid overriding existing files, in case you need to regenerate
-  // given files, delete the files first
-  if (!existsSync(newFacadeFilePath))
-    writeFileSync(join(outputFolder, `${filename}.ts`), output);
-}
-
-function generateReadmeFile(
-  readmeTemplateData: ReadmeTemplate,
-  onlyReadmeForDocs: boolean
-): void {
-  if (onlyReadmeForDocs) {
-    readmeTemplateData.exampleList.forEach((example) => {
-      // instead of relative path for docs page which will 404
-      // return the example page on the Github repo
-      example.path = `https://github.com/juju/js-libjuju/blob/master/${example.path}`;
-    });
-    Object.entries(readmeTemplateData.facadeList).forEach(
-      ([facadeName, facade]) =>
-        facade.forEach((facadeVersion) => {
-          // instead of the default (404) /api/facades/action-pruner/v1.ts
-          // return https://juju.github.io/js-libjuju/modules/facades_action_pruner_ActionPrunerV1.html
-          facadeVersion.path = `https://juju.github.io/js-libjuju/modules/facades_${facadeFolderName(
-            facadeName
-          ).replace(/-/g, "_")}_${facadeVersion.name.split(".")[0]}.html`;
-        })
-    );
-  }
-  const output: string = readmeTemplateGenerator(readmeTemplateData);
-  writeFileSync("README.md", output);
-}
-
-export function facadeFolderName(facadeName: string) {
-  // from CamelCase to kebab-case
-  return facadeName
-    .replace(/\W+/g, "-")
-    .replace(/([a-z\d])([A-Z])/g, "$1-$2")
-    .toLowerCase();
-}
-
-main();
+generator();
```

Comment:
The generate functions were moved out of this file so they could be tested with the generator actually running.